src/config/hot.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+//! Hot reloading config
+
+use crate::config::Config;
+use anyhow::{anyhow, Result};
+use notify::{Event, RecommendedWatcher, RecursiveMode, Watcher};
+use std::path::Path;
+use std::sync::mpsc::{channel, RecvTimeoutError, TryRecvError};
+use std::sync::{mpsc, Arc, RwLock};
+use std::thread;
+use std::time::Duration;
+
+pub struct HotReloadingConfig {
+    config: Arc<RwLock<ConfigWrapper>>,
+    stop_signal_tx: mpsc::Sender<()>,
+    reload_notify_rx: mpsc::Receiver<()>,
+}
+
+impl HotReloadingConfig {
+    pub fn new_with_watcher(path: String, watch_interval: u64) -> Result<Self> {
+        let (stop_signal_tx, stop_signal_rx) = channel();
+        let (reload_notify_tx, reload_notify_rx) = channel();
+
+        let config = Arc::new(RwLock::new(ConfigWrapper {
+            config: Config::from_file(path.clone())?,
+        }));
+
+        let config_cloned = config.clone();
+        thread::spawn(move || {
```

Comment:
Let's try if we can use a tokio thread so we use a green thread intead a physical one. Maybe the watcher doesn't support it but let's see if we can save some resources considering that we already have the tokio runtime.